tests/ops/integration_tests/saas/test_mailchimp_transactional_task.py

```diff
@@ -0,0 +1,214 @@
+    db,
+    consent_policy,
+    mailchimp_transactional_connection_config,
+    mailchimp_transactional_dataset_config,
+    mailchimp_transactional_identity_email,
+) -> None:
+    """Full consent request based on the Mailchimp Transactional (Mandrill) SaaS config"""
+
+    privacy_request = PrivacyRequest(
+        id=str(uuid4()),
+        consent_preferences=[{"data_use": "advertising", "opt_in": False}],
+    )
+
+    identity = Identity(**{"email": mailchimp_transactional_identity_email})
+    privacy_request.cache_identity(identity)
+
+    dataset_name = "mailchimp_transactional_instance"
+
+    v = await graph_task.run_consent_request(
+        privacy_request,
+        consent_policy,
+        build_consent_dataset_graph([mailchimp_transactional_dataset_config]),
+        [mailchimp_transactional_connection_config],
+        {"email": mailchimp_transactional_identity_email},
+        db,
+    )
+
+    assert v == {
+        f"{dataset_name}:{dataset_name}": True
+    }, "graph has one node, and request completed successfully"
+
+    execution_logs = db.query(ExecutionLog).filter_by(
+        privacy_request_id=privacy_request.id
+    )
+
+    mailchimp_transactional_logs = execution_logs.filter_by(
+        collection_name=dataset_name
+    ).order_by("created_at")
+    assert mailchimp_transactional_logs.count() == 2
+
+    assert [log.status for log in mailchimp_transactional_logs] == [
+        ExecutionLogStatus.in_processing,
+        ExecutionLogStatus.complete,
+    ]
+
+    for log in mailchimp_transactional_logs:
+        assert log.dataset_name == dataset_name
+        assert (
+            log.collection_name == dataset_name
+        ), "Node-level is given the same name as the dataset name"
+        assert log.action_type == ActionType.consent
+
+
+@pytest.mark.integration_saas
+@pytest.mark.integration_mailchimp_transactional
+@pytest.mark.asyncio
+@mock.patch("fides.api.ops.service.connectors.saas_connector.AuthenticatedClient.send")
+async def test_mailchimp_transactional_consent_prepared_requests(
+    mocked_client_send,
+    db,
+    consent_policy,
+    mailchimp_transactional_connection_config,
+    mailchimp_transactional_dataset_config,
+    mailchimp_transactional_identity_email,
+) -> None:
+    """Assert attributes of the PreparedRequest created by the client for running the consent request"""
+
+    privacy_request = PrivacyRequest(
+        id=str(uuid4()),
+        consent_preferences=[{"data_use": "advertising", "opt_in": False}],
+    )
+
+    identity = Identity(**{"email": mailchimp_transactional_identity_email})
+    privacy_request.cache_identity(identity)
+
+    await graph_task.run_consent_request(
+        privacy_request,
+        consent_policy,
+        build_consent_dataset_graph([mailchimp_transactional_dataset_config]),
+        [mailchimp_transactional_connection_config],
+        {"email": mailchimp_transactional_identity_email},
+        db,
+    )
+
+    assert mocked_client_send.called
```

Comment:
nice, this is a very thorough test and it makes the functionality easy to dissect